src/software/firmware/srcs/alarm_controller.cpp

```diff
@@ -0,0 +1,172 @@
+           * The software shall detect pressure out-of-range value in case of pressure sensor
+           * disconnection or shortcut ( <0.250v & >3.1V) and a High Priority Alarm 18 shall be
+           * triggered.
+           */
+          Alarm(AlarmPriority::ALARM_HIGH, 18u, 1u),
+
+          /**
+           * RCM-SW-11
+           * The device shall monitor the battery voltage and trig a medium priority alarm 21 when
+           * voltage is at 12.3V.
+           */
+
+          Alarm(AlarmPriority::ALARM_MEDIUM, 21u, 1u),
+
+          /**
+           * RCM-SW-14
+           * The device shall embed a medium priority alarm 22 when the Plateau pressure is not
+           * reached (absolute différence > 20% in absolute value) until the 2nd respiratory cycle.
+           */
+          Alarm(AlarmPriority::ALARM_MEDIUM, 22u, 2u),
+
+          /**
+           * RCM-SW-15
+           * The device shall embed a medium priority alarm 23 when the PEEP target is not reached
+           * (absolute difference > 2cmH2O) until the 3th respiratory cycle.
+           */
+          Alarm(AlarmPriority::ALARM_MEDIUM, 23u, 3u),
+
+          /**
+           * RCM-SW-19
+           * The device shall embed a low priority alarm 24 when the Plateau pressure < 2cm H2O at
+           * the second cycle (Patient disconnection).
+           */
+          Alarm(AlarmPriority::ALARM_LOW, 24u, 2u),
+
+          /**
+           * RCM-SW-16
+           * The device shall embed an information (audible) signal 31 when the mains are
+           * disconnected to alert the user.
+           */
+          Alarm(AlarmPriority::ALARM_LOW, 31u, 1u)}) {}
+
+void AlarmController::snooze() {
+    // TODO disable buzzer sound for 120s
+}
+
+void AlarmController::detectedAlarm(uint8_t p_alarmCode, uint32_t p_cycleNumber) {
+    for (int i = 0; i < ALARMS_SIZE; i++) {
+        Alarm* current = &m_alarms[i];
+        if (current->getCode() == p_alarmCode) {
+            current->detected(p_cycleNumber);
+            break;
+        }
+    }
+}
+
+void AlarmController::notDetectedAlarm(uint8_t p_alarmCode) {
+    for (int i = 0; i < ALARMS_SIZE; i++) {
+        Alarm* current = &m_alarms[i];
+        if (current->getCode() == p_alarmCode) {
+            current->notDetected();
+            break;
+        }
+    }
+}
+
+void AlarmController::manageAlarm() {
+    AlarmPriority highestPriority;
+    uint8_t triggeredAlarmCodes[ALARMS_SIZE];
+    uint8_t numberOfTriggeredAlarms = 0;
+
+    for (int i = 0; i < ALARMS_SIZE; i++) {
+        Alarm* current = &m_alarms[i];
+        if (current->isTriggered()) {
+            if (numberOfTriggeredAlarms == 0) {
+                highestPriority = current->getPriority();
+            }
+
+            triggeredAlarmCodes[numberOfTriggeredAlarms] = current->getCode();
+            numberOfTriggeredAlarms++;
+        }
+    }
+    displayAlarmInformation(triggeredAlarmCodes, numberOfTriggeredAlarms);
+
+    if (!highestPriority) {
```

Comment:
I think it is equivalent too highestPriority == 0, which means highestPriority == ALARM_LOW (first value of enum is 0). You can explicitely set ALARM_LOW = 1 to fix this;

Reply:
or add a ALARM_NONE = 0 value first

Reply:
Thanks. I had `ALARM_NONE`.